File: engine/engine-model/src/main/scala/tools/packet/storage/TakeOutOfStorage.scala
package tools.packet.storage

import client.inventory.{Item, MapleInventoryType}
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class TakeOutOfStorage(private var _slots: Byte, private var _inventoryType: MapleInventoryType, private var _items: java.util.Collection[Item]) extends PacketInput {
  def slots: Byte = _slots

  def inventoryType: MapleInventoryType = _inventoryType

  def items: java.util.Collection[Item] = _items

  override def opcode(): SendOpcode = SendOpcode.STORAGE
}

File: engine/engine-model/src/main/java/tools/packet/storage/TakeOutOfStorage.java
package tools.packet.storage;

import java.util.Collection;

import client.inventory.Item;
import client.inventory.MapleInventoryType;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record TakeOutOfStorage(Byte slots, MapleInventoryType inventoryType,
                               Collection<Item> items) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.STORAGE;
   }
}